docs/source/spec/http-protocol-compliance-tests.rst

```diff
@@ -0,0 +1,534 @@
+.. _http-protocol-compliance-tests:
+
+==============================
+HTTP Protocol Compliance Tests
+==============================
+
+Smithy is a protocol-agnostic IDL that tries to abstract the serialization
+format of request and response messages sent between a client and server.
+Protocol specifications define the serialization format of a protocol, and
+protocol compliance tests help to ensure that implementations correctly
+implement a protocol specification.
+
+.. contents:: Table of contents
+    :depth: 2
+    :local:
+    :backlinks: none
+
+
+--------
+Overview
+--------
+
+This specification defines two traits in the ``smithy.test`` namespace that
+are used to make assertions about client and server protocol implementations.
+
+:ref:`smithy.test#httpRequestTests <httpRequestTests-trait>`
+    Used to define how an HTTP request is serialized given a specific
+    protocol, authentication scheme, and set of input parameters.
+:ref:`smithy.test#httpResponseTests <httpResponseTests-trait>`
+   Used to define how an HTTP response is serialized given a specific
+   protocol, authentication scheme, and set of output or error parameters.
+
+Protocol implementation developers use these traits to ensure that their
+implementation is correct. This can be done through code generation of test
+cases or by dynamically loading test cases at runtime. For example, a Java
+implementation could generate JUnit test cases to assert that the
+expectations defined in a model match the behavior of a generated client
+or server.
+
+
+Parameter format
+================
+
+The ``params`` property of both the ``httpRequestTests`` trait and
+``httpResponseTests`` trait represent parameters that are used to serialize
+HTTP requests and responses. In order to compare implementation specific
+results against the expected result of each test case across different
+programming languages, parameters are defined in the same format specified
+in :ref:`trait-definition-values` with the following additional constraints:
+
+* Timestamp values must be converted to a Unix timestamp represented
+  as an integer.
+* Client implementations that automatically provide values for members marked
+  with the :ref:`idempotencyToken-trait` MUST use a constant value of
+  ``00000000-0000-4000-8000-000000000000``.
+
+
+.. _httpRequestTests-trait:
+
+----------------
+httpRequestTests
+----------------
+
+Summary
+    The ``httpRequestTests`` trait is used to define how an HTTP request is
+    serialized given a specific protocol, authentication scheme, and set of
+    input parameters.
+Trait selector
+    .. code-block:: css
+
+        operation
+Value type
+    ``object``
```

Comment:
This is a list of `httpRequestTestCase` objects - see [`@examples`](https://github.com/awslabs/smithy/blame/master/docs/source/spec/core.rst#L5039-L5042). The same fix is needed in `@httpResponseTests`.